tests/unit/test_decide_user_install.py

```diff
@@ -0,0 +1,181 @@
+    monkeypatch.setattr(EXISTS, true)
+
+
+@fixture
+def isnotdir(monkeypatch):
+    """os.path.isdir mocked to always return False."""
+    monkeypatch.setattr(ISDIR, false)
+
+
+@fixture
+def nonwritable_global(monkeypatch):
+    """test_writable_dir mocked to always return False."""
+    monkeypatch.setattr(WRITABLE, false)
+
+
+@fixture
+def writable_global(monkeypatch):
+    """test_writable_dir mocked to always return True."""
+    monkeypatch.setattr(WRITABLE, true)
+
+
+@fixture
+def virtualenv_global(monkeypatch):
+    """virtualenv_no_global mocked to always return False."""
+    monkeypatch.setattr(VIRTUALENV_NO_GLOBAL, false)
+
+
+@fixture
+def virtualenv_no_global(monkeypatch):
+    """virtualenv_no_global mocked to always return False."""
+    monkeypatch.setattr(VIRTUALENV_NO_GLOBAL, true)
+
+
+@mark.parametrize(('use_user_site', 'prefix_path', 'target_dir', 'root_path'),
+                  filter(lambda args: sum(map(bool, args)) > 1,
+                         product((False, True), (None, 'foo'),
+                                 (None, 'bar'), (None, 'baz'))))
+def test_conflicts(use_user_site, prefix_path, target_dir, root_path):
+    """Test conflicts of target, user, root and prefix options."""
+    with raises(CommandError):
+        decide_user_install(
+            use_user_site=use_user_site, prefix_path=prefix_path,
+            target_dir=target_dir, root_path=root_path)
+
+
+def test_target_exists_error(writable_global, exists, isnotdir):
+    """Test existing target which is not a directory."""
+    with raises(InstallationError):
+        decide_user_install(target_dir='bar')
+
+
+@mark.parametrize(('exist', 'is_dir'),
+                  ((false, false), (false, true), (true, true)))
+def test_target_exists(exist, is_dir, writable_global, monkeypatch):
+    """Test target paths for non-error exist-isdir combinations."""
+    monkeypatch.setattr(EXISTS, exist)
+    monkeypatch.setattr(ISDIR, is_dir)
+    assert decide_user_install(target_dir='bar') is False
+
+
+def test_target_nonwritable(nonexists, nonwritable_global):
+    """Test nonwritable path check with target specified."""
+    with raises(InstallationError):
+        decide_user_install(target_dir='bar')
+
+
+def test_target_writable(nonexists, writable_global):
+    """Test writable path check with target specified."""
+    assert decide_user_install(target_dir='bar') is False
+
+
+def test_prefix_nonwritable(nonwritable_global):
+    """Test nonwritable path check with prefix specified."""
+    with raises(InstallationError):
+        decide_user_install(prefix_path='foo')
+
+
+def test_prefix_writable(writable_global):
+    """Test writable path check with prefix specified."""
+    assert decide_user_install(prefix_path='foo') is False
+
+
+@mark.parametrize('kw', ({'use_user_site': False}, {'root_path': 'baz'}))
+def test_global_site_nonwritable(kw, nonwritable_global, virtualenv_global):
+    """Test error handling and user site-packages decision
```

Comment:
FYI this docstring is incompatible w/ PEP 257 that says that the "title" line of a docstring should fit 72 chars and end with a period. Then, you should add an empty line if you want to follow-up with a description part.

Reply:
Thanks, I forgot about that.  Though, since this is a test module, unless you feel really a strong against it, I'll probably leave it as-is.